client/qiskit_serverless/core/clients/serverless_client.py

```diff
@@ -397,9 +397,15 @@ def function(
     ####### FILES #######
     #####################
 
-    def files(self, provider: Optional[str] = None) -> List[str]:
+    def files(self, function: QiskitFunction) -> List[str]:
         """Returns list of available files produced by programs to download."""
-        return self._files_client.list(provider)
+        return self._files_client.list(function)
+
+    def provider_files(
+        self, function: QiskitFunction, provider: Optional[str] = None
```

Comment:
Technically you can obtain the title and the provider from the `QiskitFunction` instance:
- https://github.com/Qiskit/qiskit-serverless/blob/main/client/qiskit_serverless/core/function.py#L71

So you would not need to pass the provider to the function. My proposal would be to pass only the Qiskit Function in obtain the info from the instance, @korgan00 . WDYT?

And maybe we could add a check in `provider_files` to return an error if the function doesn't have a provider.